src/ctap/timed_permission.rs

```diff
@@ -163,38 +146,31 @@ mod test {
         let mut u2f_state = U2fUserPresenceState::new(request_duration(), presence_duration());
         assert!(!u2f_state.consume_up(start_time));
         assert!(u2f_state.is_up_needed(start_time));
-        // The timeout excludes equality, so it should be over at this instant.
-        assert!(!u2f_state.is_up_needed(start_time.checked_add(request_duration()).unwrap()));
     }
 
     fn grant_up_timeout(start_time: CtapInstant) {
         let mut u2f_state = U2fUserPresenceState::new(request_duration(), presence_duration());
         assert!(!u2f_state.consume_up(start_time));
         assert!(u2f_state.is_up_needed(start_time));
         u2f_state.grant_up(start_time);
-        // The timeout excludes equality, so it should be over at this instant.
-        assert!(!u2f_state.consume_up(start_time.checked_add(presence_duration()).unwrap()));
```

Comment:
Curious why this assert (and the one above) became invalid? Is this because equality is treated differently? What if you checked something like `start_time.checked_add(Milliseconds::new(1001_u64))`?

Reply:
Same remark as in the clock module regarding bound checking. We should check both sides of the bound:
- `start_time + presence_duration()`
- `start_time + presence_duration() + epsilon()` (or `- epsilon()`)